src/journal/bsod.c

```diff
@@ -188,11 +199,18 @@ static int display_emergency_message_fullscreen(const char *message) {
         if (r < 0)
                 return log_warning_errno(r, "Failed to write emergency message to terminal: %m");
 
+        char buff[STRLEN("\nScan the QR code below, and search for the entry with the ID: --") + strlen(message_id) + 1];
```

Comment:
Now, `message_id` may be NULL, so, please conditionalize the logic here and below.